library/Icingadb/Web/Controller.php

```diff
@@ -89,6 +97,39 @@ public function createPaginationControl(Paginatable $paginatable)
         $paginationControl->setDefaultPageSize($this->getPageSize(null));
         $paginationControl->setAttribute('id', $this->getRequest()->protectId('pagination-control'));
 
+        if (Url::fromRequest()->getParam(ViewModeSwitcher::DEFAULT_VIEW_MODE_PARAM) === 'minimal') {
+            $hasLimitParam = Url::fromRequest()->hasParam(LimitControl::DEFAULT_LIMIT_PARAM);
+
+            if ($paginationControl->getDefaultPageSize() <= LimitControl::DEFAULT_LIMIT && ! $hasLimitParam) {
+                $paginationControl->setDefaultPageSize($paginationControl->getDefaultPageSize() * 2);
+            }
+        }
+
+        $session = $this->Window()->getSessionNamespace(
+            $this->Window()->getId() . Url::fromRequest()->getPath()
+        );
+
+        if ($session->get('previous_page')) {
+            $prefs = $this->Auth()->getUser()->getPreferences();
+            $viewMode = $prefs->getValue('icingadb', 'view_mode');
+
+            if ($viewMode === 'minimal') {
+                $previousPage = $session->get('previous_page');
+                $limit = $paginationControl->getLimit() / 2;
+                $currentPage = (int) (floor((($previousPage * $limit) - $limit) / ($limit * 2)) + 1);
+
+                if (
+                    $paginationControl->getCurrentPageNumber() !== $currentPage
+                    && ! Url::fromRequest()->getParams()->isEmpty()
+                ) {
+                    $this->redirectNow(Url::fromRequest()->setParams([
+                        ViewModeSwitcher::DEFAULT_VIEW_MODE_PARAM => $viewMode,
+                        $paginationControl->getPageParam()        => $currentPage
+                    ]));
+                }
+            }
```

Comment:
You don't need to re-calculate the position here. If a user switches in minimal mode to another page, the previous page in common mode is invalidated.

If I'm on page 4 in common mode and switch to minimal mode, I'm on page 2. If I switch back to the common/detailed mode, I want to be on page 4 again, since that's what I've seen moments ago. But if I switch in minimal mode to page 4 and then back to common mode, I'd be surprised if I'm on page 7 after that since the list starts at another entry. I want to be on page 6 then.

So if a user changes the page in minimal mode, the previous page need to be cleared. But that I'd still want to see in `createViewModeSwitcher`. For this you just have to include the resulting page number in minimal mode (4 -> 2) in your cached data. If the page number in minimal mode changes (e.g. to 4) `4 -> 4` doesn't match.

Reply:
You're absolutely right, but those use cases you just mentioned are being computed in ``createViewModeSwitcher``. Well, that's rather for another use case, namely: let's say the user is on ``?view=common&page=4`` and switches to minimal view mode and ends up on the page ``?view=minimal&page=2``. What if the user now switches to a completely different page, e.g. from ``/icingadb/hosts -> /icingadb/services``? If he switches back to this page without touching the view mode switcher, I would expect that he ends up on the page where he was before ``/icingadb/hosts?view=minimal&page=2``.

Reply:
However, the condition is wrong. This is the correct variant.

```php
if (
         $paginationControl->getCurrentPageNumber() !== $currentPage
         && Url::fromRequest()->getParams()->isEmpty()
     ) {...}
```

Reply:
> If he switches back to this page without touching the view mode switcher, I would expect that he ends up on the page where he was before /icingadb/hosts?view=minimal&page=2.

No. Leaving the view must clear the page setting. If one wants the previous view, there's the navigation buttons to go *back*.

Reply:
✅